### PR TITLE
compare hash in chain

### LINKED ISSUE #426: Context View: Hash matched contexts do not show children
## Steps to Reproduce
- a
  - cats [cursor, context view]
- b
  - cat
    - x

## Current Behavior
When the cursor view is activated on `a.cats`, it correctly shows both contexts, but does not ~show the child `x` under `b`~ allow the cursor to be set on `b/cat~/x`. 

## Expected Behavior
`b/cat~/x` should be selectable. It work exactly the same as when the values are an exact match:

- a
  - cat
- b
  - cat
    - x

## Notes
- It has to do with the fact that `cat` and `cats` correctly hash match (i.e. `hashThought('cat') === hashThought('cats')`), but the cursor logic is not using the correct `value` in this case.
- I am not sure if the problem is in the [generation of the `thoughtsRanked` value for `setCursor`](https://github.com/cybersemics/em/blob/dev/src/components/Editable.tsx#L128-L149), or in `setCursor` itself, so I am not sure yet where the proper unit test belongs.

### PR DESCRIPTION
**Pull request by shresthabijay**
fix #426

### CODE REVIEW

**Comment by shresthabijay** [comment]
@raineorshine The previous PR #808 was too bar behind from the dev. It was faster to migrate changes using another PR. Please review. Thanks!

File: src/util/normalizeThought.ts [hunk 1]
```diff
@@ -1,31 +1,12 @@
...
-}
-
-/** Strips all html tags. */
-const stripTags = (s: string) => s.replace(REGEXP_TAGS, '')
-
-/**
- * Making character 's' will just become an empty value ''.
- * Skip it else it will cause "s" character to have same no of context as empty thoughts in the entire tree. */
-const singularize = (s: string) => s !== 's' ? pluralize.singular(s) : s
-
```
Comment (raineorshine) [inline src/util/normalizeThought.ts:28]: I would prefer to keep these functions in this module. It keeps the definitions close to their use and minimizes the surface area of the project's internal API. Then if you need to use one of these functions in a separate module then you can factor it out.

File: src/selectors/chain.ts [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
+import { Path } from '../types'
 import { State } from '../util/initialState'
-import { Path, SimplePath } from '../types'
 
 /** Merges thoughts into a context chain, removing the overlapping head. */
 // use autogenerated rank of context
-// if there is no/empty context chain, return path as-is
-const chain = (state: State, contextChain: SimplePath[], path: SimplePath): Path => {
+// if there is no/empty context chain, return thoughtsRanked as-is
+const chain = (state: State, contextChain: Path[], path: Path) => {
```
Comment (raineorshine) [inline src/selectors/chain.ts:9]: A `contextChain` should only contain `SimplePaths`, so `Path` is incorrect here. Was there a reason you needed to change the type?

File: src/selectors/chain.ts [hunk 3]
```diff
@@ -1,18 +1,18 @@
...
 
   if (!contextChain || contextChain.length === 0) return path
 
   // get the head thought in the last segment of the contextChain
   const pivot = head(contextChain[contextChain.length - 1])
-  const i = path.findIndex(child => child.value === pivot.value)
+  const i = path.findIndex(child => hashThought(child.value) === hashThought(pivot.value))
```
Comment (raineorshine) [inline src/selectors/chain.ts:15]: Let's use `normalizeThought` for comparisons since it's equivalent and faster.

File: src/selectors/chain.ts [hunk 4]
```diff
@@ -22,16 +22,16 @@ const chain = (state: State, contextChain: SimplePath[], path: SimplePath): Path
   //   console.warn('pivot', pivot)
   // }
 
-  const append = path.slice(i - 1) as SimplePath
+  const append = path.slice(i - 1)
   const contexts = getContextsSortedAndRanked(state, pivot.value)
-  const appendedThoughtInContext = contexts.find(child => head(child.context) === append[0].value)
+  const appendedThoughtInContext = contexts.find(child => hashThought(head(child.context)) === hashThought(append[0].value))
```
Comment (raineorshine) [inline src/selectors/chain.ts:27]: Let's use `normalizeThought` for comparisons since it's equivalent and faster.

**Review by raineorshine — CHANGES_REQUESTED** [review]

File: src/util/normalizeThought.ts (hunk 1 above)
Comment (shresthabijay) [reply src/util/normalizeThought.ts:28]: Understood.

File: src/selectors/chain.ts (hunk 2 above)
Comment (shresthabijay) [reply src/selectors/chain.ts:9]: Ahh yes! I made a mistake here. Thanks for pointing it out.

File: src/util/normalizeThought.ts [hunk 5]
```diff
@@ -1,30 +1,30 @@
...
+import * as pluralize from 'pluralize'
 import { REGEXP_TAGS } from '../constants'
-
-/** Converts a string to lowecase. */
-const lower = (s: string) => s.toLowerCase()
+import emojiStrip from 'emoji-strip'
 
 /** Trims a string. */
-const trim = (s: string) => s.replace(
+export const trim = (s: string) => s.replace(
```
Comment (raineorshine) [inline src/util/normalizeThought.ts:7]: These are internal API and should not be exported, but not a big issue.

**Review by raineorshine — APPROVED** [review]

**Pull request merged**